### ISSUE TITLE
Report error when accessing generic instance variable from class object

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7051]
The typing spec indicates that if an instance variable within a generic class is generic, an attempt to access it through the class object should generate an error because the type is erased at runtime for the class.

```python
from typing import Generic, TypeVar

T = TypeVar("T")

class Node(Generic[T]):
    label: T

    def __init__(self, label: T | None = None) -> None:
        ...

Node[int].label = 1  # Should be type error
Node[int].label  # Should be type error
Node.label  # Should be type error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.348, which was just published. It will also be included in a future release of pylance.
